### ISSUE TITLE
Real-time pitch adjustment

### ISSUE THREAD

**Issue by svgeesus** [WebAudio/web-audio-api-v2#14]
Split out from https://github.com/WebAudio/web-audio-api/issues/938 as a separate issue because this is orthogonal to on-the-fly decoding of compressed loop buffers.

**Comment by svgeesus** [comment]
[A Brief History of Pitch Correction in Music](https://www.izotope.com/en/blog/music-production/a-brief-history-of-pitch-correction-in-music.html)

**Comment by jas-ableton** [comment]
Yes, this would be very useful to have built into web audio sample playback. Being able to vary the playback speed while maintaining a constant pitch, or vary pitch while maintaining a constant playback speed would allow developers to implement lots of very useful workflows that are found in common DAWs (synchronizing recorded audio to a timeline, for example).

In a DAW like Ableton Live, this concept is referred to as "warping": https://www.ableton.com/en/manual/audio-clips-tempo-and-warping/

**Comment by padenot** [comment]
The variety of possible approaches makes this a good candidate to have (a) prototype(s) using `AudioWorklet`, to investigate the space in terms of algorithm (PSOLA/WSOLA/phase vocoder, a million others), and API.

This would probably be a new node to have maximal flexibility.

**Comment by rtoy** [comment]
I think we also agreed that it should be a processing node and not a source node.  You feed it some audio, and it will pitch adjust the input.

**Comment by rtoy** [comment]
TPAC 2020:

@padenot said this has been implemented in an AudioWorklet.  Next step is to get the code and see how it works and propose a new node and API for this.
